core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -509,6 +511,11 @@ final class ShuffleBlockFetcherIterator(
     (currentResult.blockId, new BufferReleasingInputStream(input, this))
   }
 
+  def toCompletionIterator: Iterator[(BlockId, InputStream)] = {
+    CompletionIterator[(BlockId, InputStream), this.type](this,
```

Comment:
Although `CompletionIterator` frees the reference to the `ShuffleBlockFetcherIterator` in `ShuffleFetchCompletionListener`, doesn't this `CompletionIterator` also internally keep another reference to the `ShuffleBlockFetcherIterator`? Will it be released early too?

Reply:
@viirya 
It will be released early in `CoalecedRDD`, for it's just one of the iterators for `CoalescedRDDPartition`, once it's exhausted, there are no objects will refer to it, so it can be released.

Reply:
As it is exhausted, I think it is almost the time the task finishes. So I'm wondering how much the benefit it gets.

Reply:
In CoaleaseRDD, a task reads many iterators. So when one iterator exhausted, the task may still have many more iterators to consume.

Reply:
oh, I see. Thanks.